src/main/java/com/example/resilience4jdemo/helpers/HttpClient.java

```diff
@@ -7,8 +7,19 @@
 import reactor.core.publisher.Mono;
 
 /** @author RishabhSairawat */
+/**
+ * A utility class for making HTTP requests.
+ */
 public class HttpClient {
 
+  /**
+   * Sends a GET request to the specified URI and retrieves the response body as a Mono.
+   *
+   * @param uriString      the URI string to send the request to
+   * @param typeReference  the parameterized type reference representing the expected response type
+   * @param <T>            the type of the response body
+   * @return a Mono representing the response body
+   */
   public static <T> Mono<T> genericGet(
       String uriString, ParameterizedTypeReference<T> typeReference) {
     UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromUriString(uriString);
```

Comment:
**Suggestion:** Consider handling exceptions that may occur during the HTTP request. This can improve the robustness of your code and provide more informative error messages to the caller. [best practice]
```suggestion
  public static <T> Mono<T> genericGet(
      String uriString, ParameterizedTypeReference<T> typeReference) {
    try {
      UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromUriString(uriString);
    } catch (Exception e) {
      throw new RuntimeException("Failed to send GET request to " + uriString, e);
    }
  }
```